crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1846,7 +1852,13 @@ impl<'db> TypeInferenceBuilder<'db> {
 
                 unpacked.get(name_ast_id).unwrap_or(Type::Unknown)
```

Comment:
It would be unusual, but I think valid according to the spec to do `FOO, BAR = ...` in a stub file. So I think we also should add a special case for `...` (only if `file.is_stub(db)` of course) in `infer_unpack_types` (where we infer the `value_ty` from the RHS expression). And a test for this.

Reply:
I applied this change inside the unpacker since that was the first place that value type was being inferred. Also there was a match to check the expression being used. I hope this is what you meant as well.

But if I understand correctly we don't want the unpacking to happen in for loops. So I added a flag to unpacker to only add this case to assignments.